### PR TITLE
rust: fix glibc build

### PR DESCRIPTION
**Pull request by LnL7**
###### Motivation for this change

Tested rustc, cargo and a few rust packages.

/cc @vcunat

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by erictapen** [comment]
There currently are two more approaches to fix `rustc` breakage, discussed in #34189;
* Patching `glibc`
* Adjusting the `rustc` test in question to expect the result happening.

File: pkgs/development/compilers/rust/rustc.nix [hunk 1]
```diff
@@ -101,6 +101,10 @@ stdenv.mkDerivation {
     # Useful debugging parameter
     # export VERBOSE=1
   ''
+  + optionalString (stdenv ? glibc) ''
+    # Disable test for glibc-2.26
```
Comment (vcunat) [inline pkgs/development/compilers/rust/rustc.nix]: Nitpick: it's patched glibc-2.25.

**Comment by vcunat** [comment]
I think disabling the whole test as in this PR is the best way, at least for now.

**Comment by LnL7** [comment]
Yeah, unless it really is a glibc bug ofcorse but doesn't look like that's the case.

**Comment by erictapen** [comment]
One thing that irritates me is that the test [remains unchanged in newest `rustc` version](https://github.com/rust-lang/rust/blob/766bd11c8a3c019ca53febdcd77b2215379dd67d/src/test/run-pass/out-of-stack.rs). How can that compile? Anyway, I think disabling the test is absolutely sufficient for `release-17.09`.

**Review by vcunat — APPROVED** [review]

**Comment by vcunat** [comment]
Strange, but I see no use in waiting any more.  We can improve later, when/if we find out more.

**Comment by erictapen** [comment]
Oh of course! I'm happy there is a solution to the issue in. Thanks for merging.

**Pull request merged**